content/chainguard/chainguard-enforce/how-to-install-chainctl.md

```diff
@@ -84,17 +84,231 @@ You should receive output similar to the following.
+
+```sh
+SIGNATURE_URL="https://dl.enforce.dev/chainctl/${CHAINCTL_VERSION?}/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m).sig"
+CERTIFICATE_URL="https://dl.enforce.dev/chainctl/${CHAINCTL_VERSION?}/chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m).cert.pem"
+```
+
+Check your environment variables for correctness using `echo`:
+
+```sh
+echo "$SIGNATURE_URL\n$CERTIFICATE_URL\n$CHAINCTL_VERSION"
+```
+
+You should receive output like the following, which shows the correct version, operating system, and CPU architecture for your system values in the computed URLs:
+
+```
+https://dl.enforce.dev/chainctl/0.1.42/chainctl_darwin_arm64.sig
+https://dl.enforce.dev/chainctl/0.1.42/chainctl_darwin_arm64.cert.pem
+0.1.42
+```
+
+Now you can verify the binary by running the following `cosign verify-blob` command:
+
+```sh
+COSIGN_EXPERIMENTAL=1 cosign verify-blob --signature $SIGNATURE_URL --certificate $CERTIFICATE_URL $(which chainctl)
+```
+
+You should receive the following output:
+
+```
+tlog entry verified with uuid: 1a481ce6eeee6306dffbba9fef701a21623ca0b07780b32b009395c905f7df7a index: 8959219
+Verified OK
+```
+
+If you the signature or certificate URLs are incorrect, if there is a problem with your `chainctl` binary, or if your `$CHAINCTL_VERSION` doesn't match the version output by `chainctl version --json`, you will receive an error like the following:
+
+```
+Error: verifying blob [/usr/local/bin/chainctl]: invalid signature when validating ASN.1 encoded signature
+. . .
+```
+
+Check the environment variables that you set for correctness, and ensure that the path reported by `$(which chainctl)` points to the version of `chainctl` that you downloaded and installed. 
+
+For completeness, you can use the following script to check `chainctl` binaries using Cosign. It is a combination of the previous steps into one script with some additional checks for missing tools, and error handling.
+
+```bash
+#!/bin/bash
+set -Eeuo pipefail
+export COSIGN_EXPERIMENTAL=1
+
+trap 'cmd_error $LINENO "${BASH_COMMAND}"' ERR
+function cmd_error() {
+  echo "error running command on line $1"
+  echo "command is: $2"
+}
+
+function check_signature {
+    chainctl_version=$(chainctl version 2>&1 |awk '/GitVersion/ {print $2}')
+    chainctl_os_arch="chainctl_$(uname -s | tr '[:upper:]' '[:lower:]')_$(uname -m)"
+    sig_url="https://dl.enforce.dev/chainctl/${chainctl_version?}/${chainctl_os_arch?}.sig"
+    cert_url="https://dl.enforce.dev/chainctl/${chainctl_version?}/${chainctl_os_arch?}.cert.pem"
+    cosign verify-blob --signature "${sig_url?}" --cert "${cert_url?}" "$(command -v chainctl)"
+}
+
+function check_executable {
+    status=0
+    executable_name="${1?}"
+    command -v "${executable_name?}" > /dev/null || status=$?
+    if [[ "${status?}" -ne 0 ]]; then
+        echo "${executable_name?} not found"
+        exit 1
+    fi
+}
+
+check_executable "cosign"
+check_executable "chainctl"
+check_signature
+```
+
```

Comment:
I like to add the sections with a sentence so that the user feels like the section is over.

```suggestion
This script offers a combination of the previous steps into one script with some additional checks for missing tools, and error handling.

```